### ISSUE TITLE
[UX] Reword "Use HTTPS for canonical URLs"

### ISSUE THREAD

**Issue by klonos** [backdrop/backdrop-issues#4195]
This is a follow-up of #868 (that issue was closed as a duplicate of #2871).

@docwilmot:
> Great that this is moving along. I Googled what canonical means, and I kinda get that its a good idea. I would ask though that we find a simpler easy to grasp description to go with this option, or change the label itself from `[ ] Redirect to canonical URL` to something simple. I'd give suggestions but I've forgotten again what exactly it means.

@quicksketch:
>How about:
>
>```
>[ ] Redirect to clean or aliased URLs if available
>```
>
>After thinking about this functionality and the possible trouble you could get into with it, I don't think we should try to handle HTTP/HTTPS redirects, or any redirects involving domain name changes (e.g. www/non-www or old-domain.com to new-domain.com). The potential for redirecting to broken URLs or loops conflicting with .htaccess redirects seems too high. Redirecting to clean or aliased versions only seems a much safer and more common issue. So in a way, renaming this option to avoid the term "canonical" (which would also deal with domain name and protocol) makes more sense to me anyway.

---
PR by @klonos: ...

**Comment by jenlampton** [comment]
> Check if the https variable is already set in settings.php; if not, disable that checkbox.

I don't think this setting means what you think it means. This setting should have no affect on URLs.

The `https` value in `settings.php` is for Mixed-mode sessions:  `Set to TRUE to create both secure and insecure sessions when using HTTPS.`

This would allow someone to log in on a `http` page, then click through to a `https` page, and remain logged in. On most sites, a session created for `http` would NOT work on `https`.

**Comment by alanmels** [comment]
Bit off-topic, but in the light of [HTTPS Everywhere](https://www.eff.org/https-everywhere) has started to widely adopted by all the major browsers and the hosting industry, IMO HTTPS settings should be preferred in settings, even made default, leaving users an option to choose HTTP if needed.

If it's not too bold with too much possible complications, Backdrop could declare itself as HTTPS Everywhere CMS. We do it anyway on hosting level as excerpt from https://www.altagrade.com/help/control-panel/site-manager says:

> **SSL Certificate**
> AltaGrade is a hosting vendor fully adopted the HTTPS-everywhere practice. With all the major browsers starting to enforce HTTPS, it is becoming imperative to have HTTPS-compatible websites, which in turn requires to have properly installed SSL certificates for them. So you have several options on your AltaGrade server's control panel for SSL certifications when creating new domain names:

**Comment by olafgrabienski** [comment]
> This is a follow-up of #868 (that issue was closed as a duplicate of #2871).

Oh, there is already a follow-up. Let's take it as an reminder to fix the original issue (Redirect to clean or aliased URLs if available)!

**Comment by BWPanda** [comment]
Is this a duplicate of https://github.com/backdrop/backdrop-issues/issues/4865, or vice versa?
